### PR TITLE
[chainservice] delete APIServer interface

### PR DESCRIPTION
**Pull request by huof6829**
# Description

delete APIServer interface

Fixes https://github.com/iotexproject/iotex-core/issues/3246

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: chainservice/chainservice.go [hunk 1]
```diff
@@ -138,14 +137,8 @@ func (cs *ChainService) ActionPool() actpool.ActPool {
...
-type APIServer interface {
-	GetActions(ctx context.Context, in *iotexapi.GetActionsRequest) (*iotexapi.GetActionsResponse, error)
-	GetReceiptByAction(ctx context.Context, in *iotexapi.GetReceiptByActionRequest) (*iotexapi.GetReceiptByActionResponse, error)
-}
-
-// APIServer returns the API server
-func (cs *ChainService) APIServer() APIServer {
+// APIServer returns the grpc server
+func (cs *ChainService) APIServer() *api.GRPCServer {
 	return cs.api.GrpcServer
```
Comment (Liuhaai) [inline chainservice/chainservice.go:142]: I think `CoreService` Interface should be returned here

Comment (huof6829) [reply chainservice/chainservice.go:142]: if return CoreService interface,  the final called function [GetActionByActionHash](https://github.com/iotexproject/iotex-core/blob/c39701e6baa2ad5a06eaaa3c4f307db7699f767a/tools/util/injectorutil.go#L841)  and [GetReceiptByAction](https://github.com/iotexproject/iotex-core/blob/c39701e6baa2ad5a06eaaa3c4f307db7699f767a/tools/util/injectorutil.go#L854) will convert CoreService interface to GrpcService.  Can this be better ?

Comment (huof6829) [reply chainservice/chainservice.go]: Done. Thanks.

File: e2etest/rewarding_test.go [hunk 2]
```diff
@@ -239,7 +239,7 @@ func TestBlockEpochReward(t *testing.T) {
 	rps := make([]*rewarding.Protocol, numNodes)
 	sfs := make([]factory.Factory, numNodes)
 	chains := make([]blockchain.Blockchain, numNodes)
-	apis := make([]chainservice.APIServer, numNodes)
+	coreServices := make([]api.CoreService, numNodes)
```
Comment (Liuhaai) [inline e2etest/rewarding_test.go:242]: revert

File: e2etest/rewarding_test.go [hunk 3]
```diff
@@ -259,7 +259,7 @@ func TestBlockEpochReward(t *testing.T) {
 		sfs[i] = svrs[i].ChainService(configs[i].Chain.ID).StateFactory()
 
 		chains[i] = svrs[i].ChainService(configs[i].Chain.ID).Blockchain()
-		apis[i] = svrs[i].ChainService(configs[i].Chain.ID).APIServer()
+		coreServices[i] = svrs[i].ChainService(configs[i].Chain.ID).APIServer()
```
Comment (Liuhaai) [inline e2etest/rewarding_test.go:262]: revert

File: api/serverV2.go [hunk 4]
```diff
@@ -158,3 +158,8 @@ func (svr *ServerV2) Stop(ctx context.Context) error {
 func (svr *ServerV2) ReceiveBlock(blk *block.Block) error {
 	return svr.core.ReceiveBlock(blk)
 }
+
+// CoreService returns the member
```
Comment (Liuhaai) [inline api/serverV2.go:162]: `CoreService returns the coreservice of the api`

**Review by Liuhaai — APPROVED** [review]

File: api/serverV2.go [hunk 5]
```diff
@@ -158,3 +158,8 @@ func (svr *ServerV2) Stop(ctx context.Context) error {
 func (svr *ServerV2) ReceiveBlock(blk *block.Block) error {
 	return svr.core.ReceiveBlock(blk)
 }
+
+// CoreService returns the coreservice of the api
+func (svr *ServerV2) CoreService() CoreService {
+	return svr.core
```
Comment (dustinxie) [inline api/serverV2.go:164]: i don't see you delete `APIServer` in chainservice.go?

Comment (huof6829) [reply api/serverV2.go:164]: this is done by  https://github.com/iotexproject/iotex-core/pull/3259

**Review by dustinxie — APPROVED** [review]

**Pull request merged**